### PR TITLE
[ads] Send HTTP response status for landed events as part of the confirmation token redemption

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/36752

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Confirm http response status is sent in confirmation token redemptions for page lands if the target url fails to load such as 404 NOT FOUND etc.

### CODE REVIEW

**Comment by tmancey** [comment]
Draft until we rebase once https://github.com/brave/brave-core/pull/22574 has been merged

File: components/brave_ads/core/internal/account/confirmations/user_data_builder/confirmation_user_data_builder.cc [hunk 1]
```diff
@@ -24,12 +23,8 @@ void BuildFixedUserDataCallback(base::Value::Dict user_data,
   UserDataInfo confirmation_user_data;
 
   confirmation_user_data.dynamic = std::move(dynamic_user_data);
-
   confirmation_user_data.fixed = std::move(fixed_user_data);
-
-  if (UserHasJoinedBraveRewards()) {
-    confirmation_user_data.fixed.Merge(std::move(user_data));
-  }
+  confirmation_user_data.fixed.Merge(std::move(user_data));
```
Comment (tmancey) [inline components/brave_ads/core/internal/account/confirmations/user_data_builder/confirmation_user_data_builder.cc:27]: This is data passed as part of `DepositWithUserData` and will be guarded on a per use case.

File: components/brave_ads/core/internal/ad_units/user_data/page_land_user_data.cc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+namespace {
+constexpr char kHttpResponseStatusCodeKey[] = "httpStatusResponseCode";
+}  // namespace
+
+base::Value::Dict BuildPageLandUserData(const TabInfo& tab) {
+  base::Value::Dict user_data;
+
+  if (tab.http_response_status_code != net::HTTP_OK) {
+    user_data.Set(kHttpResponseStatusCodeKey, tab.http_response_status_code);
+  }
```
Comment (tmancey) [inline components/brave_ads/core/internal/ad_units/user_data/page_land_user_data.cc:22]: Adds HTTP status response code to the confirmation token redemption payload

**Review by aseren — APPROVED** [review]

**Review by kylehickinson — APPROVED** [review]
iOS++

**Pull request merged**